msr.h

```diff
@@ -78,6 +79,10 @@ class SafeMsrHandle
 
     int32 read(uint64 msr_number, uint64 * value)
     {
+        if (!AllowMSRAccess(msr_number))
```

Comment:
there is high chance that the whitelist is not complete on some of the platforms especially for PCICFG registers (PCM supports many platforms). This change will make the code silently fail and/or deliver corrupted values if the list is not complete. Instead it should be very explicit about the failed white-list check printing which access has been denied (address, etc.) and return 0 (not (int32)sizeof(uint64) - size of bytes read successfully). This will allow to quickly root-cause the issue and complete the white-list.